src/components/Trade/TradeInput.tsx

```diff
@@ -415,37 +408,19 @@ export const TradeInput = () => {
     [sellAmountCryptoPrecision, buyAmountCryptoPrecision, isTradeQuotePending],
   )
 
-  // TODO(woodenfurniture): this needs to be rewritten for arbitrary assets to support multi-hop
   const hasSufficientProtocolFeeBalances = useMemo(() => {
-    if (protocolFees === undefined) return false
-
-    const buyAssetTradeFeeCryptoBaseUnit = protocolFees[buyAsset.assetId]?.requiresBalance
-      ? protocolFees[buyAsset.assetId].amountCryptoBaseUnit
-      : '0'
-
-    const sellAssetTradeFeeCryptoBaseUnit = protocolFees[sellAsset.assetId]?.requiresBalance
-      ? protocolFees[sellAsset.assetId].amountCryptoBaseUnit
-      : '0'
+    if (protocolFees === undefined || sellAssetAccountId === undefined) return false
 
-    const feeAssetTradeFeeCryptoBaseUnit =
-      feeAsset && protocolFees[feeAsset.assetId]?.requiresBalance
-        ? protocolFees[feeAsset.assetId].amountCryptoBaseUnit
-        : '0'
+    const sellAccountBalancesBaseUnit = portfolioAccountBalancesBaseUnit[sellAssetAccountId]
 
-    return (
-      bn(buyAssetBalanceCryptoBaseUnit).gte(buyAssetTradeFeeCryptoBaseUnit) &&
-      bn(sellAssetBalanceCryptoBaseUnit).gte(sellAssetTradeFeeCryptoBaseUnit) &&
-      bn(feeAssetBalanceCryptoBaseUnit).gte(feeAssetTradeFeeCryptoBaseUnit)
-    )
-  }, [
-    buyAsset.assetId,
-    buyAssetBalanceCryptoBaseUnit,
-    feeAsset,
-    feeAssetBalanceCryptoBaseUnit,
-    protocolFees,
-    sellAsset.assetId,
-    sellAssetBalanceCryptoBaseUnit,
-  ])
+    // protocol fees that require balance are always paid from the sell account
```

Comment:
Can we add a comment bringing a bit more nuance in what we refer to as account (number? actual `AccountId`?) and explaining that this is very much an oversimplification for the time being, until we plumb introspecting buy asset balance through?

Protocol fees are *generally* paid from the sell account (as in, the account *number*, but also chain/asset of the sell asset), though they may not, i.e ATOM -> OSMO where a fee needs to be paid for the first hop in ATOM, and for the last hop, in OSMO.
In this case, the buy asset account number may or may not be the same as the sell asset account number, but most importantly, protocol fees for each would be denominated for a whole different chain/asset. Sure, the Osmosis swapping implementation is turbo wrong with intermediary hops, but even if/when we implement it with a single swap the way osmosis.zone does, the last hop fee that should be paid would still be in OSMO, which is a different account.

Reply:
excellent callout, actioned.